### ISSUE TITLE
redirects do not work from within a back end widget

### ISSUE THREAD

**Issue by fritzmg** [contao/core-bundle#976]
In an extension of mine I use
```php
\Controller:redirect(…);
```
within a back end Widget for a Facebook Login button, to redirect to the Facebook Login url. However, in Contao 4 this does not work anymore, due to the new handling of redirects with Exceptions. The back end widget will simply show this error now instead of the redirect being executed:
```
This exception has no message. Use $exception->getResponse() instead.
```
![screen-shot-2017-07-21-at-11 57 30](https://user-images.githubusercontent.com/4970961/28458969-e7d65bf6-6e0b-11e7-92b5-2867ade11a2e.png)

**Comment by fritzmg** [comment]
nvm, error in my extension :) (had a try catch block)

**Comment by Toflar** [comment]
We could add an exception to the exception here https://github.com/contao/core-bundle/blob/master/src/Resources/contao/classes/DataContainer.php#L428

Not sure if `RedirectException` is the only one though. Maybe you can make a PR to never catch a `RedirectException` but also maybe have this somehow configurable so that any other developer can at least fix it by adding the config? Would that be an idea?

**Comment by leofeyer** [comment]
So far, we are only catching the `RedirectResponseException`.

https://github.com/contao/core-bundle/blob/master/src/EventListener/PrettyErrorScreenListener.php#L203

I think this is the only exception type that behaves differently, so we don't have to catch other types IMO.

**Comment by Toflar** [comment]
I agree, I just want to make sure we don't trap developers. Configuring should be an easy pick :)

**Comment by aschempp** [comment]
Why do we not catch ResponseException (which is a parent of RedirectResponseException)?

**Comment by leofeyer** [comment]
Fixed in 950d1972de383534f3c6165099ed216cd0a8190e.
